packages/adapter-flagsmith/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import type { Adapter } from 'flags';
+import flagsmith from 'flagsmith';
+import { IIdentity, IInitConfig, IFlagsmithFeature } from 'flagsmith/types';
+
+export type { IIdentity } from 'flagsmith/types';
+export { getProviderData } from './provider';
+
+let defaultFlagsmithAdapter: AdapterResponse | undefined;
+
+export type FlagsmithValue = IFlagsmithFeature['value'];
+
+export type EntitiesType = IIdentity;
+
+export type AdapterResponse = {
+  booleanValue: () => Adapter<boolean, EntitiesType>;
+  stringValue: () => Adapter<string, EntitiesType>;
+  numberValue: () => Adapter<number, EntitiesType>;
+};
+
+function createFlagsmithAdapter(params: IInitConfig): AdapterResponse {
+  async function initialize() {
+    await flagsmith.init({ fetch: globalThis.fetch, ...params });
```

Comment:
Since flagsmith is a singleton, does this mean there can only ever be one instance of the adapter at the same time?

If it gets called twice to connect to different projects the first adapter would get reconfigured, right?

That could be quite surprising, but I'm fine with this if we document it

```
createFlagsmithAdapter()
createFlagsmithAdapter()
```